e2e/test/iothub/service/DirectMethodE2ETests.cs

```diff
@@ -0,0 +1,136 @@
+        public async Task DirectMethodsClient_DeviceOnline_DeviceClientNotOpen()
+        {
+            // arrange
+            IotHubServiceClient serviceClient = TestDevice.ServiceClient;
+            await using TestDevice testDevice = await TestDevice.GetTestDeviceAsync(_devicePrefix);
+
+            IotHubDeviceClient deviceClient = testDevice.CreateDeviceClient();
+            await testDevice.OpenWithRetryAsync().ConfigureAwait(false);
+            var methodInvocation = new DirectMethodServiceRequest("someDirectMethod");
+
+            // ensure device client is closed
+            await deviceClient.CloseAsync();
+            
+            try
+            {
+                // act
+                Func<Task> act = async () => await serviceClient.DirectMethods.InvokeAsync(testDevice.Id, methodInvocation);
+
+                // assert
+                ExceptionAssertions<IotHubServiceException> errorContext = await act.Should().ThrowAsync<IotHubServiceException>();
+                errorContext.And.StatusCode.Should().Be(System.Net.HttpStatusCode.NotFound);
+            }
+            finally
+            {
+                await deviceClient.CloseAsync();
+            }
+        }
+
+        [TestMethod]
+        [Timeout(TestTimeoutMilliseconds)]
+        public async Task DirectMethodsClient_DeviceOnline_MethodCallbackTimesOut()
+        {
+            // arrange
+            IotHubServiceClient serviceClient = TestDevice.ServiceClient;
+            await using TestDevice testDevice = await TestDevice.GetTestDeviceAsync(_devicePrefix);
+            var methodInvocation = new DirectMethodServiceRequest("someDirectMethod")
+            {
+                ConnectionTimeout = TimeSpan.FromSeconds(3),
+                ResponseTimeout = TimeSpan.FromSeconds(3),
+            };
+            IotHubDeviceClient deviceClient = testDevice.CreateDeviceClient();
+            await deviceClient.OpenAsync();
+            await testDevice.OpenWithRetryAsync().ConfigureAwait(false);
+
+            try
+            {
+                // act
+                await deviceClient
+                .SetDirectMethodCallbackAsync(
+                    (methodRequest) =>
+                    {
+                        // force a timeout
+                        Thread.Sleep(10000);
+                        var response = new DirectMethodResponse(200);
+                        return Task.FromResult(response);
+                    })
+                .ConfigureAwait(false);
+
+                // assert
+                Func<Task> act = async() => await serviceClient.DirectMethods.InvokeAsync(testDevice.Id, methodInvocation);
+                ExceptionAssertions<IotHubServiceException> response = await act.Should().ThrowAsync<IotHubServiceException>();
+                response.And.StatusCode.Should().Be(System.Net.HttpStatusCode.BadRequest);
+            }
+            finally
+            {
+                await deviceClient.CloseAsync();
```

Comment:
await using on the TestDevice will close and dispose the device client, so you can get rid of the try/finally block.